language/go/generate.go

```diff
@@ -415,7 +415,9 @@ func (g *generator) generateProto(mode proto.Mode, target protoTarget, importPat
 	goProtoLibrary.SetAttr("proto", ":"+protoName)
 	g.setImportAttrs(goProtoLibrary, importPath)
 	if target.hasServices {
-		goProtoLibrary.SetAttr("compilers", []string{"@io_bazel_rules_go//proto:go_grpc"})
+		goProtoLibrary.SetAttr("compilers", []string{g.c.GoGrpcCompiler})
+	} else {
+		goProtoLibrary.SetAttr("compilers", []string{g.c.GoProtoCompiler})
```

Comment:
`compilers` shouldn't be set if it's only `@io_bazel_rules_go//proto:go_proto` (even if that value was set explicitly).

Reply:
I'd worry a bit about the case where `rules_go` changes the default (to e.g. `:gofast_proto`) but an older `gazelle` is still in use; in that case, passing `-go_proto_compiler @io_bazel_rules_go//proto:go_proto` would mean you confusingly get `:gofast_proto`, rather than `:go_proto` (since we'd omit `compilers`).